src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1206,8 +1206,15 @@ private int checkAndAddEntry(int pos, int index)
                 entries[index++] = hash;
                 entries[index++] = next;
                 entries[index  ] = pos;
+                // Validate comment if it exists
+                int clen = CENCOM(cen, pos);
+                if (clen > 0) {
+                    int elen = CENEXT(cen, pos);
+                    int start = entryPos + nlen + elen;
+                    zcp.toString(cen, start, clen);
```

Comment:
The change look okay but you may want to put another comment on zcp.toString to say that it may throw, otherwise something looking at the code might wonder if it can be removed as the returned String is not used.